gradle-module-plugin/README.md

```diff
@@ -148,3 +155,6 @@ the result.
 * v0.1.0-SNAPSHOT-6 - changed how credentials and files are specified for signing and publication.  The keys are the same, but properties are now expected to exist in a gradle.properties file, or to be specified as runtime flags as described in the Usage section above.
 * v0.1.0-SNAPSHOT-12 - added checksum generation and build report tasks.  Split repo into separate builds using gradle build composition to better isolate changes.  
 * v0.1.0-SNAPSHOT-15 - fixed dependency collection, renamed 'AssembleModuleAssets' task class and associated task
+* v0.1.0-SNAPSHOT-16 - changed modlImplementation configuration to not resolve transitive dependencies
+* v0.1.0-SNAPSHOT-17 - modlImplementation and modlApi configurations have been replaced with a single `modlDependency` configuration to simplify dependency marking and avoid confusing differences between compile-time and modl runtime environments, 
+* v0.1.0-SNAPSHOT-18 - reverted to separate modlImplementation and modlApi configurations.  However, modlImplementation retains the same resolution semantics as gradle's `implementation`, while also fully resolving `modlImplementation`'s transitive dependencies for inclusion into the modl.  This change results in logical handling of dependencies in IDE/gradle environments, while also ensuring that needed dependencies are bundled in the module for use a runtime.
```

Comment:
I think the notes here are still great to include for our own internal consumption about the expected behavior or advanced users.